### ISSUE TITLE
HTC Vive Steam VR theatre problems - Euro Truck Sim 2 & Dirt Rally no longer working in VR

### ISSUE THREAD

**Issue by GTGeoff** [LibreVR/Revive#473]
Hi
 
I seem to be having problem with Steam VR theatre and would appreciate some help please?
 
I’ve had Euro Truck Sim 2 working fine and Dirt Rally running fine via Revive.
 
Euro truck was working for weeks when suddenly it seemed to stop going into theatre mode as normal, I pressed F11 but the game just worked on the desktop and not VR anymore. I re-downloaded the game. Still no joy.
 
Dirt Rally is now the same, even when using the latest version of Revive and latest injectors I put files in and all was working ok, then in the last couple of weeks it just doesn’t go into theatre like normal. The game loads fine out of VR but via VR now it brings up a message ‘drt.exe file isn’t executable goes black then shuts down.
 
I re-downloaded the game and verified game files none of which makes a difference.
 
I tried Raceroom last night, it begins to work normally in theatre mode then it goes back to the ‘this real’ and nothing happens in theatre, it just shows on desktop only.
 
BTW Assetto Corsa still works fine and all my other VR games.
 
Any idea’s what could be the problem?

**Comment by ecnahc** [comment]
Well I was playing Euro Truck Sim 2 just last week, because it supports the Vive now in the Oculus build. I just went to the Beta's tab and chose the one that says Oculus/Vive, then added =openvr to the launch commands I think. Revive wasn't necessary. As for DIRT, I was about to purchase it on sale but your post makes me thing I should wait.

**Comment by GTGeoff** [comment]
Yes it’s strange as I had euro truck 2 working fine in VR using that method but then suddenly it stopped, then later Dirt rally stopped after it was running fine in VR.
 
I tried project cars and that works, (although I don’t play it much due to poor performance).
 
I just wonder if it was a windows 10 update or graphics driver update that’s caused the problem.

**Comment by ecnahc** [comment]
Well Euro Truck doesn't need Revive anymore, so maybe removing any of the patch files and typing in launch option -openvr will start Euro Truck for you. Try switching from steamvr to steamvr beta, or vise versa. All kinds of weird stuff can get these games working. Restarting everything is important though.

**Comment by GTGeoff** [comment]
Ok thanks, I’ll try that.
 
Dirt Rally did work for a while so maybe it’s just something random on my PC and it might be ok for you? It’s a shame Codemasters just don’t add native VR support, I’m sure people would happily pay for a DLC if it’s about cost.

**Comment by TiE23** [comment]
I'll have to go and try more things myself but I'm also having trouble with Dirt Rally. I last recall having it work on Christmas Day and yesterday it would crash on launch. I believe Dirt Rally has not been patched since then, so I'm not sure if it's "just us" with Geoff, a SteamVR issue, a Revive issue, or some nVidia issue (980Ti).

Just making this comment so it's not immediate closed.

**Comment by GTGeoff** [comment]
Ok, yes it’s crash at launch for me. It may be coincidence but it seemed to happen after I updated to the latest nVidia driver (Titan X - 2015)

**Comment by adamb70** [comment]
I'm getting the same issue with Dirt, anybody found a solution yet?

**Comment by TiE23** [comment]
Don't know what changed but tonight I was able to get it running and play a few stages before it crashed the game when my stations lost sync for a moment (eh, it happens). Didn't change anything other than re-confirming I had used the revive Patch... tool like always with the drt.exe file. It doesn't seem that there has been any updates with SteamVR... http://store.steampowered.com/news/?appids=250820 (as far as I can tell).

**Comment by Balderick** [comment]
GPU driver updates?

**Comment by adamb70** [comment]
It's working for me now too, and I haven't installed the latest Nvidia drivers yet, I'm still on 378.57.

I'm not sure what the change is, it looks like [SteamVR ](https://steamdb.info/app/250820/depots/) hasn't been updated since I tried 4 days ago, and neither has [DiRt](https://steamdb.info/app/310560/depots/).

**Comment by Armada651** [comment]
Dirt Rally is working fine for me, though there seem to be some more problems with the current patching method (see issue #399). I will probably replace the patching method with an actual launcher sometime, but that might take a while.

**Comment by GTGeoff** [comment]
Hi

I've now got it working again. Its a little bit embarrassing but I hadn't right clicked the revive icon on the button right and selected 'patch' then the dirt executable file. 

It's there on the instructions so my bad. It works perfectly now. 

Just posting to apologise and in case anyone else missed this like I did. 

Great job with revive and the latest update is definitely allowing the game to run better. 

Thank you. 

Geoff
